apps/aeutils/test/aeu_env_tests.erl

```diff
@@ -10,17 +10,30 @@
 
 -include_lib("eunit/include/eunit.hrl").
 
-all_test_() ->
+schema_test_() ->
     {setup,
      fun setup/0,
      fun teardown/1,
-     [{"Example configuration tests",
+     [{"Example user configuration files pass schema validation",
        [fun() ->
           ?assertMatch({ok, _}, aeu_env:check_config(Config))
         end || Config <- test_data_config_files()]
        }]
     }.
 
+extra_checks_test_() ->
+    {setup,
+     fun() -> ok = meck:new(setup, [passthrough]), setup() end,
+     fun(R) -> teardown(R), ok = meck:unload(setup) end,
+     [{"Example user configuration files pass checks further to the schema", %% For enabling files to be linked from wiki as examples.
+       [fun() ->
+                {ok, {UserMap, UserConfig}} = aeu_env:check_config(Config),
+                ok = mock_user_config(UserMap, UserConfig),
+                ?assertEqual(ok, aec_hard_forks:check_env())
```

Comment:
I'm not too much into the test suites, but I guess this is a "unit test"? Isn't calling additional module an antipattern ?
Also mocking in unit tests.

Reply:
Mocking in unit tests is fine I think, it's about isolating the "unit". However, I still think mocking in general is an anti pattern and that interfaces that are unit tested should be structured in such a way that mocking is not needed (parameterization, dependency injection, pure functional code etc.)